src/main/java/me/ars/pokerbot/IrcBot.java

```diff
@@ -27,20 +28,24 @@
    */
   private final Set<String> admins = new HashSet<>();
   private Roster roster;
+  private final IrcBotConfig config;
 
-  public IrcBot(String startingChannel) {
+  public IrcBot(IrcBotConfig config) {
     try {
       roster = Roster.getRoster();
     } catch (IOException e) {
       e.printStackTrace();
     }
     tables = new HashMap<>();
+    this.config = config;
+    final String startingChannel = config.irc.channel;
     final IrcStateCallback startingCallback = new IrcStateCallback(startingChannel);
-    tables.put(startingChannel, new Table(startingCallback, roster));
+    tables.put(startingChannel, new Table(startingCallback, roster, config.game));
 
-    setName(Constants.BOT_NAME);
+    final String botName = config.irc.nick;
```

Comment:
Can the nick or any of the other config options be null at this time?

Reply:
No, the configuration file is assumed to be fully loaded before IrcBot (or other future frontends) instanciates.